js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx

```diff
@@ -109,13 +111,19 @@ export const AppProvider = (props: AppProviderProps) => {
   const retryLink = React.useMemo(() => {
     return new RetryLink({
       attempts: {
-        max: 2,
-        retryIf: (error, _operation) => {
-          return error && error.statusCode && [502, 503, 504].includes(error.statusCode);
+        max: 3,
+        retryIf: async (error, _operation) => {
+          if (error && error.statusCode && httpStatusCodesToRetry.includes(error.statusCode)) {
+            return true;
+          }
+          return false;
         },
       },
-      delay: {
-        initial: 300,
+
+      delay: (_retryCount, _operation, error) => {
+        // Retry-after header is in seconds, concert to ms by multiplying by 1000.
+        const wait = parseFloat(error?.response?.headers?.get?.('retry-after') ?? '0.3') * 1000;
```

Comment:
Make `'0.3'` a const so that it's a named value?

Reply:
Should we add logic that uses _retryCount to cap this behavior ? With the new 0.3s default behavior, this will retry a 429 indefinitely and I worry that if the user is navigating around the app and initiating additional requests that 429 they'll stack into a huge volume.

Reply:
The retryCount here is meant to influence the delay for example to do exponential backoff. Once you're in this function though there's no way to stop the retry. To stop the retry based off count we specified "max"